test/queue-test.js

```diff
@@ -648,17 +640,31 @@ describe('Queue', (it) => {
         redisScanCount: 50
       });
 
-      queue.process(async () => {});
 
       // Choose a big number for numbers of jobs created, because otherwise the
       // set will be encoded as an intset and SSCAN will ignore the COUNT param.
       // https://redis.io/commands/scan#the-count-option
+      const createJobPromises = [];
       for (let i = 0; i < 10000; i++) {
-        await queue.createJob({foo: 'bar'}).save();
+        createJobPromises.push(queue.createJob({foo: 'bar'}).save());
       }
+      Promise.all(createJobPromises);
+
+      // Wait for all jobs to process to make sure the SET encoding is a hash table
+      // rather than an intset.
+      const allJobsProcessed = new Promise((resolve) => {
```

Comment:
switch this to use `reef` (there are examples of it being used with `process`) - ask me if you need clarification